core/subsurfacestartup.c

```diff
@@ -295,7 +295,7 @@ void setup_system_prefs(void)
 	default_prefs.font_size = system_divelist_default_font_size;
 
 #if !defined(SUBSURFACE_MOBILE)
-	default_prefs.default_filename = system_default_filename();
+	default_prefs.default_filename = copy_string(system_default_filename());
```

Comment:
actually, this creates a memory leak :-)
Read the system_default_filename() function. That string is malloc()ed there every time you call that function.

Reply:
https://github.com/Subsurface-divelog/subsurface/blob/master/core/windows.c#L114
```
	static const char *path = NULL;
	if (!path)
		path = system_default_path_append(filename);
	return path;
```
it's a static pointer.